Mathlib/RingTheory/Polynomial/Selmer.lean

```diff
@@ -57,14 +57,14 @@ theorem X_pow_sub_X_sub_one_irreducible (hn1 : n ≠ 1) : Irreducible (X ^ n - X
   apply IsUnitTrinomial.irreducible_of_coprime' ⟨0, 1, n, zero_lt_one, hn, -1, -1, 1, rfl⟩
   rintro z ⟨h1, h2⟩
   apply X_pow_sub_X_sub_one_irreducible_aux z
-  rw [trinomial_mirror zero_lt_one hn (-1 : ℤˣ).ne_zero (1 : ℤˣ).ne_zero] at h2
-  simp_rw [trinomial, aeval_add, aeval_mul, aeval_X_pow, aeval_C,
-    Units.val_neg, Units.val_one, map_neg, map_one] at h1 h2
-  replace h1 : z ^ n = z + 1 := by linear_combination h1
-  replace h2 := mul_eq_zero_of_left h2 z
-  rw [add_mul, add_mul, add_zero, mul_assoc (-1 : ℂ), ← pow_succ, Nat.sub_add_cancel hn.le] at h2
-  rw [h1] at h2 ⊢
-  exact ⟨rfl, by linear_combination -h2⟩
+  · rw [trinomial_mirror zero_lt_one hn (-1 : ℤˣ).ne_zero (1 : ℤˣ).ne_zero] at h2
+    simp_rw [trinomial, aeval_add, aeval_mul, aeval_X_pow, aeval_C,
+      Units.val_neg, Units.val_one, map_neg, map_one] at h1 h2
+    replace h1 : z ^ n = z + 1 := by linear_combination h1
+    replace h2 := mul_eq_zero_of_left h2 z
+    rw [add_mul, add_mul, add_zero, mul_assoc (-1 : ℂ), ← pow_succ, Nat.sub_add_cancel hn.le] at h2
+    rw [h1] at h2 ⊢
+    exact ⟨rfl, by linear_combination -h2⟩
```

Comment:
I don't understand this change: there doesn't seem to be an alternative goal here?

Reply:
The extra goal is figuring out what `n` should be.  It is the exponent of `z` that you see on line 63, that gets unified only on the penultimate line, with the `rw`.

I decided to give the hint to `apply` that `(n := n)` to avoid the extra goal and the indentation.